packages/experimental/src/components/StatusIcon/StatusIcon.js

```diff
@@ -0,0 +1,193 @@
+  CheckmarkFilled32,
+  InformationSquareFilled16,
+  InformationSquareFilled20,
+  InformationSquareFilled24,
+  InformationSquareFilled32,
+  Renew16,
+  Renew20,
+  Renew24,
+  Renew32,
+  Time16,
+  Time20,
+  Time24,
+  Time32,
+} from '@carbon/icons-react';
+
+const icons = [
+  {
+    type: 'fatal',
+    small: <Misuse16 />,
+    medium: <Misuse20 />,
+    large: <Misuse24 />,
+    'x-large': <Misuse32 />,
+  },
+  {
+    type: 'critical',
+    small: <ErrorFilled16 />,
+    medium: <ErrorFilled20 />,
+    large: <ErrorFilled24 />,
+    'x-large': <ErrorFilled32 />,
+  },
+  {
+    type: 'major-warning',
+    small: <WarningAltInvertedFilled16 />,
+    medium: <WarningAltInvertedFilled20 />,
+    large: <WarningAltInvertedFilled24 />,
+    'x-large': <WarningAltInvertedFilled32 />,
+  },
+  {
+    type: 'minor-warning',
+    small: <WarningAltFilled16 />,
+    medium: <WarningAltFilled20 />,
+    large: <WarningAltFilled24 />,
+    'x-large': <WarningAltFilled32 />,
+  },
+  {
+    type: 'undefined',
+    small: <UndefinedFilled16 />,
+    medium: <UndefinedFilled20 />,
+    large: <UndefinedFilled24 />,
+    'x-large': <UndefinedFilled32 />,
+  },
+  {
+    type: 'unknown',
+    small: <UnknownFilled16 />,
+    medium: <UnknownFilled20 />,
+    large: <UnknownFilled24 />,
+    'x-large': <UnknownFilled32 />,
+  },
+  {
+    type: 'normal',
+    small: <CheckmarkFilled16 />,
+    medium: <CheckmarkFilled20 />,
+    large: <CheckmarkFilled24 />,
+    'x-large': <CheckmarkFilled32 />,
+  },
+  {
+    type: 'info',
+    small: <InformationSquareFilled16 />,
+    medium: <InformationSquareFilled20 />,
+    large: <InformationSquareFilled24 />,
+    'x-large': <InformationSquareFilled32 />,
+  },
+  {
+    type: 'in-progress',
+    small: <Renew16 />,
+    medium: <Renew20 />,
+    large: <Renew24 />,
+    'x-large': <Renew32 />,
+  },
+  {
+    type: 'running',
+    small: <Renew16 />,
+    medium: <Renew20 />,
+    large: <Renew24 />,
+    'x-large': <Renew32 />,
+  },
+  {
+    type: 'pending',
+    small: <Time16 />,
+    medium: <Time20 />,
+    large: <Time24 />,
+    'x-large': <Time32 />,
+  },
+];
+
+import { pkg } from '../../settings';
+
+import cx from 'classnames';
+
+export const StatusIcon = ({ type, theme, size, className, ...rest }) => {
+  const [icon, setIcon] = useState([]);
+  const [iconSize, setIconSize] = useState('');
+  const [iconTheme, setIconTheme] = useState(null);
+
+  const classNames = cx({
+    [`${pkg.prefix}-status-icon--${iconTheme}`]: iconTheme,
+    [`${pkg.prefix}-status-icon--${iconTheme}-${type}`]: type,
+    className,
+  });
+
+  useEffect(() => {
+    type && setIcon(...icons.filter((icon) => icon.type === type));
+  }, [type]);
+
+  useEffect(() => {
+    size && setIconSize(size);
+  }, [size]);
+
+  useEffect(() => {
+    theme && setIconTheme(theme);
+  }, [theme]);
+
+  return (
+    <div className={classNames} {...rest}>
+      {icon && iconTheme && iconSize ? icon[`${iconSize}`] : null}
```

Comment:
This is the final part of the suggestion above to stash the components in an object rather than render functions. Here, finally, if and when we have an icon component identified, we create a render function for it and insert it into our rendered output.

```suggestion
      {IconComponent && <IconComponent />}
```